### PR TITLE
ipfs: 0.4.20 -> 0.4.21 (module, drop deps.nix)

### PR DESCRIPTION
**Pull request by dtzWill**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/networking/ipfs/default.nix [hunk 1]
```diff
@@ -1,21 +1,21 @@
-{ stdenv, buildGoPackage, fetchFromGitHub }:
+{ stdenv, buildGoModule, fetchFromGitHub }:
 
-buildGoPackage rec {
+buildGoModule rec {
   name = "ipfs-${version}";
```
Comment (teto) [inline pkgs/applications/networking/ipfs/default.nix]: buildGoModule won't accept `pname =` ?

**Review by fpletz — APPROVED** [review]

File: pkgs/applications/networking/ipfs/default.nix (hunk 1 above)
Comment (kalbasit) [reply pkgs/applications/networking/ipfs/default.nix]: it does.

Comment (dtzWill) [reply pkgs/applications/networking/ipfs/default.nix]: Yes it does, I just missed it :innocent: .  Updated accordingly, thanks!

**Comment by dtzWill** [comment]
Should be good now, thanks!

**Comment by lifranc** [comment]
Getting a build error for modules here.
```sh
go: finding github.com/go-toolsmith/astcast v0.0.0-20181028201508-b7a89ed70af1
go: error loading module requirements
builder for '/nix/store/00qvnbnjd2w86jpqvgj79gzxji6swjy2-ipfs-0.4.21-go-modules.drv' failed with exit code 1
```

**Pull request merged**